utilities/setup_dev_bucket.py

```diff
@@ -0,0 +1,351 @@
+import boto3
+import importlib
+import os
+from pathlib import Path
+import typer
+from typing import Literal
+
+PROD_RECIPES_BUCKET = "edm-recipes"
+PROD_PUBLISHING_BUCKET = "edm-publishing"
+os.environ["RECIPES_BUCKET"] = PROD_RECIPES_BUCKET
+os.environ["PUBLISHING_BUCKET"] = PROD_PUBLISHING_BUCKET
+
+from dcpy import configuration
+from dcpy.models.connectors.edm.recipes import DatasetKey
+from dcpy.models.connectors.edm.publishing import (
+    ProductKey,
+    BuildKey,
+    DraftKey,
+    PublishKey,
+)
+from dcpy.utils import s3
+from dcpy.utils.logging import logger
+from dcpy.connectors.edm import recipes, publishing
+from dcpy.lifecycle.builds import plan
+
+
+ROOT_PATH = Path(__file__).parent.parent
+BUCKET_PREFIX = "de-dev-"
+PRODUCTS = [
+    "db-cbbr",
+    "db-colp",
+    "db-cpdb",
+    "db-developments",
+    "db-facilities",
+    "db-green_fast_track",
+    "db-template",
+    "db-pluto",
+    "db-zoningtaxlots",
+]
+
+
+def copy_folder(
```

Comment:
seems like a nice s3 util? Also, should you use a `tempfile` folder locally, rather than `./tmp/dev_bucket_setup`?

Reply:
Well there is an s3 for copying a folder, but it does it without download/uploading. And while it offers ability to copy across buckets, this apparently isn't a guarantee that it could work (given the issues I ran into)

So I should probably
- edit existing function in `s3` to not have `target_bucket` as argument
- add this function to `s3` as `copy_folder_across_buckets` or something like that. This could even attempt to copy just using the s3 api to avoid downloading, and fallback to downloading.

And definitely use a tempfile folder, yes

Thoughts?

Reply:
Totally makes sense.